supervisely/api/project_api.py

```diff
@@ -969,6 +982,275 @@ def update_custom_data(self, id: int, data: Dict, silent: bool = False) -> Dict:
         )
         return response.json()
 
+    def get_custom_data(self, id: int) -> Dict[Any, Any]:
+        """Returns custom data of the Project by ID.
+        Custom data is a dictionary that can be used to store any additional information.
+
+        :param id: Project ID in Supervisely.
+        :type id: int
+        :return: Custom data of the Project
+        :rtype: :class:`dict`
+
+        :Usage example:
+
+        .. code-block:: python
+
+            import supervisely as sly
+
+            api = sly.Api.from_env()
+
+            project_id = 123456
+
+            custom_data = api.project.get_custom_data(project_id)
+
+            print(custom_data) # Output: {'key': 'value'}
+        """
+        return self.get_info_by_id(id).custom_data
+
+    def _get_system_custom_data(self, id: int) -> Dict[Any, Any]:
+        """Returns system custom data of the Project by ID.
+        System custom data is just a part of custom data that is used to store system information
+        and obtained by the key `_METADATA_SYSTEM_KEY`.
+
+        :param id: Project ID in Supervisely.
+        :type id: int
+        :return: System custom data of the Project
+        :rtype: :class:`dict`
+
+        :Usage example:
+
+        .. code-block:: python
+
+            import supervisely as sly
+
+            api = sly.Api.from_env()
+
+            project_id = 123456
+
+            system_custom_data = api.project._get_system_custom_data(project_id)
+
+            print(system_custom_data)
+        """
+        return self.get_info_by_id(id).custom_data.get(_METADATA_SYSTEM_KEY, {})
+
+    def get_validation_schema(self, id: int, use_caching: bool = False) -> Optional[Dict[Any, Any]]:
+        """Returns validation schema of the Project by ID.
+        Validation schema is a dictionary that can be used to validate metadata of each entity in the project
+        if corresnpoding schema is provided.
+        If using caching, the schema will be loaded from the cache if available.
+        Use cached version only in scenarios when the schema is not expected to change,
+        otherwise it may lead to checks with outdated schema.
+
+        :param id: Project ID in Supervisely.
+        :type id: int
+        :param use_caching: If True, uses cached version of the schema if available.
+            NOTE: This may lead to checks with outdated schema. Use with caution.
+            And only in scenarios when the schema is not expected to change.
+        :return: Validation schema of the Project
+        :rtype: :class:`dict`
+
+        :Usage example:
+
+        .. code-block:: python
+
+            import supervisely as sly
+
+            api = sly.Api.from_env()
+
+            project_id = 123456
+
+            validation_schema = api.project.get_validation_schema(project_id)
+
+            print(validation_schema) # Output: {'key': 'Description of the field'}
+        """
+        SCHEMA_DIFF_THRESHOLD = 60 * 60  # 1 hour
```

Comment:
maybe do this as an argument?

Reply:
@almazgimaev, since it's a custom feature and there are a lot of arguments already, this was made on purpose.